### DISCUSSION TITLE
How do I scroll on touch-screens?

### DISCUSSION THREAD

**Discussion by JanDiederich** [slint-ui/slint#3254] [Q&A] [answered]
I have a `ScrollView`, which can't be scrolled when used on Touch-Screens.
If I try panning gestures, nothing happens. I tried adding the gestures manually, like in an official example, but that works only for mouse pointers.

If I try to run it on a touch screen, I can't really scroll, more, like a tiny amount, than it stops. Tested with a touch screen on Raspberry Pi OS 64 Bit with the latest updates. I see no mouse pointer on the touch screen, only if I connect a mouse, that mouse panning then behaves like on a AMD64 Ubuntu 20 64 Bit. But touch doesn't work.

Code:
```
timespan_scroll := ScrollView {
    in-out property<length> value <=> self.viewport-y;
    
    // Vertical layout is absolutely needed to inform ScrollView how to scroll.
    text-layout := VerticalLayout {
        Text {
             ...
        }
    }
           
   temp_touch := TouchArea {
      property <length> pressed-value;

      pointer-event(event) => {
          /*Removing or adding this doesn't change a thing: event.button == PointerEventButton.left && */
          if (event.kind == PointerEventKind.down) {                                            
              self.pressed-value = timespan_scroll.value;
          }
      }
      moved => {
          if (self.enabled && self.pressed) {
              timespan_scroll.value = 
                  // Don't scroll more down than the list is high.
                  max(-(text-layout.height - timespan_scroll.height),
                  // Don't scroll more up than topmost.
                  min(0, self.pressed-value + (self.mouse-y - self.pressed-y)));
          }
      }
  }
```

How can I pan on touch screens?

**Comment by ogoffart** [comment, marked as answer]
The current ScrollView is meant for desktop and it's true that it isn't touch friendly (we actually should do better than that).

If you want something can can pan on touch, you can use Flickable instead

**Reply by JanDiederich** [reply]
Please add this to your official documentation. I can't find any mention of that important point [there](https://slint.dev/releases/1.1.1/docs/slint/src/builtins/widgets#scrollview).
